src/NHibernate/Logging.cs

```diff
@@ -53,29 +107,49 @@ public interface IInternalLogger2
 		bool IsEnabled(InternalLogLevel logLevel);
 	}
 
+	// Since 5.1
 	[Obsolete("Implement INHibernateLoggerFactory instead")]
 	public interface ILoggerFactory
 	{
 		IInternalLogger LoggerFor(string keyName);
 		IInternalLogger LoggerFor(System.Type type);
 	}
 
+	/// <summary>
+	/// Factory interface for providing a <see cref="INHibernateLogger"/>.
+	/// </summary>
 	public interface INHibernateLoggerFactory
 	{
-		IInternalLogger2 LoggerFor(string keyName);
-		IInternalLogger2 LoggerFor(System.Type type);
+		/// <summary>
+		/// Get a logger for the given log key.
+		/// </summary>
+		/// <param name="keyName">The log key.</param>
+		/// <returns>A NHibernate logger.</returns>
+		INHibernateLogger LoggerFor(string keyName);
+		/// <summary>
+		/// Get a logger using the given type as log key.
+		/// </summary>
+		/// <param name="type">The type to use as log key.</param>
+		/// <returns>A NHibernate logger.</returns>
+		INHibernateLogger LoggerFor(System.Type type);
 	}
 
-	public class LoggerProvider
+	/// <summary>
+	/// Provide methods for getting NHibernate loggers according to supplied <see cref="INHibernateLoggerFactory"/>.
+	/// </summary>
+	/// <remarks>
+	/// By default, it will use a <see cref="Log4NetLoggerFactory"/> if log4net is available, otherwise it will
+	/// use a <see cref="NoLoggingLoggerFactory"/>.
+	/// </remarks>
+	public static class LoggerProvider
```

Comment:
Switched to static: constructor was private (so not a breaking change) and it simplifies the implementation.